### PR TITLE
Add Window CI using Azure pipelines

### PR DESCRIPTION
**Pull request by fsfod**
Add a CI for windows using [Azure pipelines](https://azure.microsoft.com/en-gb/services/devops/pipelines/) which is free for open source projects and has GitHub integration. I have it setup using my own account here [here](https://tfransham.visualstudio.com/LuaJIT/_build?definitionId=1&_a=summary) but it would be better to make your own account to run it off.
The azure-pipelines.yml file has a build matrix setup to compile on windows for both 32 bit, 64 bit and GC 64 mode using MSVC 2019 compiler. The msvcbuild.bat is very limited in what build options you can pass it on the command line so windows is not compiled with asserts unlike Linux

### CODE REVIEW

**Comment by siddhesh** [comment]
Thanks, I looked around but couldn't determine if Azure pipelines has support for architectures like arm64 and ppc64le.  I suppose we'll also have to add travis into the mix to get a wider testing matrix.

File: linux.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+steps:
+  - script: |
+     cd src
+     make -j  XCFLAGS="$(buildFlags)"
+    displayName: Linux Build
+
+  - script: |
+     cd test
+     ../src/luajit test.lua
```
Comment (siddhesh) [inline linux.yml]: You can use the `make check` target to do this.

**Comment by siddhesh** [comment]
I've added Azure Pipelines for this project now, so once you replace the manual invocation of test.lua with `make check` it's good for a merge.

**Comment by fsfod** [comment]
> 
> 
> Thanks, I looked around but couldn't determine if Azure pipelines has support for architectures like arm64 and ppc64le. I suppose we'll also have to add travis into the mix to get a wider testing matrix.

You can run docker images on azure pipeline which people have used to run other arch tests.

**Pull request merged**